apps/andi/lib/andi_web/input_schemas/ingestion_metatadata_form_schema.ex

```diff
@@ -54,4 +62,19 @@ defmodule AndiWeb.InputSchemas.IngestionMetadataFormSchema do
     |> AtomicMap.convert(safe: false, underscore: false)
     |> changeset()
   end
+
+  defp validate_top_level_selector(%{changes: %{sourceFormat: source_format}} = changeset)
+       when source_format in ["xml", "text/xml"] do
+    validate_required(changeset, [:topLevelSelector], message: "is required")
```

Comment:
XML question(I rarely use XML) - Why is a topLevelSelector required? Isn't it similar to JSON and could automatically just parse the first element, allowing overwritting topLevelSelector to start at a nested field?

Reply:
A good question that I'm not sure the answer to. All XML examples in our system use a topLevelSelector, and it's been a required field since I've been involved.

As we get feedback on ingesting xml datasets, this could definitely be a potential improvement.